core/alerts/alerts-core/src/main/java/com/thinkbiganalytics/alerts/api/core/AggregatingAlertProvider.java

```diff
@@ -239,7 +239,18 @@ public boolean removeAlertManager(AlertManager mgr) {
         AlertSource src = getSource(alertId.sourceId);
 
         if (src != null) {
-            return getAlert(alertId.alertId, src);
+            return getAlert(alertId.alertId, src,false);
+        } else {
+            return null;
+        }
+    }
+
+    public Optional<Alert> getAlertAsServiceAccount(ID id) {
+        SourceAlertID alertId = asSourceAlertId(id);
+        AlertSource src = getSource(alertId.sourceId);
+
+        if (src != null) {
+            return getAlert(alertId.alertId, src,true);
         } else {
             return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Methods with an "Optional" return type should never return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2789)